tests/phpunit/tests/formatting/linksAddBaseUrl.php

```diff
@@ -0,0 +1,67 @@
+ <?php
+
+/**
+ * Tests for the links_add_base_url function.
+ *
+ * @group formatting
+ *
+ * @covers ::links_add_base_url
+ */
+class Tests_formatting_linksAddBaseUrl extends WP_UnitTestCase {
+
+	/**
+	 * @ticket 60389
+	 *
+	 * @dataProvider data_links_add_base_url
+	 */
+	public function test_links_add_base_url( $content, $base, $attrs, $expected ) {
+		if( $attrs ) {
+			$this->assertSame( $expected, links_add_base_url( $content, $base, $attrs ) );
+		} else {
+			$this->assertSame( $expected, links_add_base_url( $content, $base ) );
+		}
+
+	}
+
+	public function data_links_add_base_url() {
+		return array(
```

Comment:
Some additional cases that I think need to be covered:

- `http` rather than `https`
- making sure that unless `data-url` is set in the `attrs`, it isn't adjusted
- the same as what is the current `data_url` but with one of them setup properly
- no protocol URLs present in content (`//localhost/url` )
- URLs that use other protocols like FTP and mailto
- A data URL in one of the attributes

I have no idea what the results for some of these are going to be.

Reply:
added some more test